File: 프로그래머스/lv2/42584. 주식가격/주식가격.py
def solution(prices):
    answer = []
    for i in range(len(prices)):
        cnt = 0
        if i == len(prices)-1:
            answer.append(cnt)
        else:
            st = prices[i]
            flag = 0 # 도중에 끊기는지 끝까지 가는지
            for j in range(i+1,len(prices)):
                # print(i,j)
                if st <= prices[j]:
                    cnt += 1
                else:
                    flag = 1
                    break
            if flag == 1:
                cnt += 1             
            answer.append(cnt)
            
    return answer

File: 프로그래머스/2/42584. 주식가격/주식가격.java
import java.util.*;
class Solution {
    public List<Integer> solution(int[] prices) {
        List<Integer> answer = new ArrayList<>();
        for(int i=0;i<prices.length;i++){
            int cnt=0;
            for(int j=i;j<prices.length;j++){
                if (prices[i] > prices[j] || j == prices.length-1) {
                    answer.add(cnt);
                    break;
                }
                else{
                    cnt++;
                }
            }
            // System.out.println(prices[i]);
        }
        return answer;
    }
}